modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/integration/ViewsIntegrationTest.java

```diff
@@ -0,0 +1,432 @@
+        assertViewsCount("MY_SCHEMA", "MY_VIEW2", 1);
+
+        assertRowsCount(5, "select * from my_view");
+        assertRowsCount(4, "select * from cache1.my_view");
+        assertRowsCount(3, "select * from cache2.my_view");
+        assertRowsCount(2, "select * from my_schema.my_view1");
+        assertRowsCount(1, "select * from my_schema.my_view2");
+
+        // PUBLIC.MY_VIEW
+        assertEquals(5, cache1.query(new SqlFieldsQuery("select * from public.my_view")).getAll().size());
+        assertEquals(5, cache1.query(new SqlFieldsQuery("select * from my_view").setSchema("PUBLIC")).getAll().size());
+        // CACHE1.MY_VIEW
+        assertEquals(4, cache1.query(new SqlFieldsQuery("select * from my_view")).getAll().size());
+        assertEquals(4, cache2.query(new SqlFieldsQuery("select * from cache1.my_view")).getAll().size());
+        // CACHE2.MY_VIEW
+        assertEquals(3, cache2.query(new SqlFieldsQuery("select * from my_view")).getAll().size());
+        assertEquals(3, cache1.query(new SqlFieldsQuery("select * from my_view").setSchema("CACHE2")).getAll().size());
+        // MY_SCHEMA.MY_VIEW1
+        assertEquals(2, cache3.query(new SqlFieldsQuery("select * from my_view1")).getAll().size());
+        assertEquals(2, cache4.query(new SqlFieldsQuery("select * from my_view1")).getAll().size());
+        // MY_SCHEMA.MY_VIEW2
+        assertEquals(1, cache3.query(new SqlFieldsQuery("select * from my_view2")).getAll().size());
+        assertEquals(1, cache4.query(new SqlFieldsQuery("select * from my_view2")).getAll().size());
+    }
+
+    /**
+     * Tests views drop on schema drop.
+     */
+    @Test
+    public void testSchemaCleaning() throws Exception {
+        initGrids(3);
+        initTable(5);
+
+        client.getOrCreateCache(new CacheConfiguration<>("CACHE1")
+            .setIndexedTypes(Integer.class, Integer.class));
+
+        client.getOrCreateCache(new CacheConfiguration<>("CACHE2")
+            .setIndexedTypes(Integer.class, Integer.class)
+            .setSqlSchema("PUBLIC")
+        );
+
+        client.getOrCreateCache(new CacheConfiguration<>("CACHE3")
+            .setIndexedTypes(Integer.class, Integer.class)
+            .setSqlSchema("MY_SCHEMA")
+        );
+
+        client.getOrCreateCache(new CacheConfiguration<>("CACHE4")
+            .setIndexedTypes(Integer.class, String.class)
+            .setSqlSchema("MY_SCHEMA")
+        );
+
+        sql("create or replace view my_view as select * from my_table");
+        sql("create or replace view cache1.my_view as select * from public.my_table");
+        sql("create or replace view my_schema.my_view as select * from public.my_table");
+
+        assertViewsCount("PUBLIC", "MY_VIEW", 1);
+        assertViewsCount("CACHE1", "MY_VIEW", 1);
+        assertViewsCount("MY_SCHEMA", "MY_VIEW", 1);
+
+        client.destroyCache("CACHE1");
+
+        assertViewsCount("PUBLIC", "MY_VIEW", 1);
+        assertViewsCount("CACHE1", "MY_VIEW", 0);
```

Comment:
Why we remove a schema when it contains a view which requests an existing tabe/cache? There a VIEW which calls existing `public.my_table`

Reply:
We don't have dedicated schema management procedures, except cache create/destroy. When schema is dropped (force by cache destroy) - it's assumed that schema objects is dropped too.